src/coreclr/jit/emitarm64.cpp

```diff
@@ -4732,19 +4732,19 @@ void emitter::emitIns_R_R(
             assert(isVectorRegister(reg1));
             assert(isVectorRegister(reg2));
 
-            if (isValidVectorDatasize(size))
+            if (insOptsAnyArrangement(opt))
             {
                 // Vector operation
-                assert(insOptsAnyArrangement(opt));
+                assert(isValidVectorDatasize(size));
                 assert(isValidArrangement(size, opt));
                 elemsize = optGetElemsize(opt);
                 fmt      = IF_DV_2M;
             }
             else
             {
-                NYI("Untested");
                 // Scalar operation
```

Comment:
Since you are updated the scalar variant `CMEQ Vd, Vn, #0` - can you add the corresponding emitter test?

Reply:
Makes sense to me, I'll add one.